### PR TITLE
chore: treeshakable

### LINKED ISSUE #138: feat(tes & design): improve bundling
## Description

Right now Design & TES are using tsup to generate the build the problem is that there is only one file generated globally in the build. Apply tree shaking to it so that the user can import the individual components & reduce there overall build size & loading time. Tree shaking is a popular solution for that & can be implemented in tsup like this https://tsup.egoist.dev/#tree-shaking

### Tasks
- [x]  Code splitting and declaration files generation -  #167 
- [ ] Add shoelace assets entry point
- [ ] Build cdn package
- [ ] Add support for React

### PR DESCRIPTION
**Pull request by JaeAeich**
## Description
Implement tree shaking in tsup bundling.

### CODE REVIEW

**Comment by JaeAeich** [comment]
@git-anurag-hub The docs just mention adding `--treeshakable` to the build command. I have tried it but there weren't many changes I could notice in the build files. I tried changing the import method as well.

**Comment by SalihuDickson** [comment]
@JaeAeich, this PR hasn't solved the problem if implementing tree shaking in the bundled package.

**Comment by SalihuDickson** [comment]
@JaeAeich, basically the reason this doesn't achieve what we're trying to do is it that this flag enables tree shaking during the bundling process which basically just removes dead code (code that isn't being used) while bundling. What we want to achieve is code splitting to make the package tree shakeable so that when a developer uses this package they can choose to only bundle the pieces of code they use and they won't be forced to bundle the entire package.

**Comment by JaeAeich** [comment]
@git-anurag-hub Honestly no! I don't have a specific good stance for doing this, but I was studying other repos and lit.dev and this was what I found to be recommended. Both are doing the same thing (its mentioned on lit) but this way we reduce the number of files and imports. Also I believe we would want to have different exports and having multiple index.ts just for barrel exports wasn't making sense. And I read this way apparently will help ts undestand types better.

`We recommend adding an HTMLElementTagNameMap entry for all elements authored in TypeScript, and ensuring you publish your .d.ts typings in your npm package.` [refer](https://lit.dev/docs/components/defining/)

Lastly I think this comes down to how we agree to write the code I don;t think there is a benifit over the other, we can remove it if we think there isn't much diff. Was just testing things as we don't have a large code base yet.

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 1]
```diff
@@ -406,3 +406,8 @@ export class Form extends LitElement {
     `;
   }
 }
+declare global {
+  interface HTMLElementTagNameMap {
+    "ecc-utils-design-form": Form;
+  }
+}
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply packages/ecc-utils-design/src/components/form/form.ts:413]: @git-anurag-hub Honestly no! I don't have a specific good stance for doing this, but I was studying other repos and lit.dev and this was what I found to be recommended. Both are doing the same thing (its mentioned on lit) but this way we reduce the number of files and imports. Also I believe we would want to have different exports and having multiple index.ts just for barrel exports wasn't making sense. And I read this way apparently will help ts undestand types better.

`We recommend adding an HTMLElementTagNameMap entry for all elements authored in TypeScript, and ensuring you publish your .d.ts typings in your npm package.` [refer](https://lit.dev/docs/components/defining/)

Lastly I think this comes down to how we agree to write the code I don;t think there is a benifit over the other, we can remove it if we think there isn't much diff. Was just testing things as we don't have a large code base yet.

**Comment by JaeAeich** [comment]
> Hey @JaeAeich, the deployment for the docs is not working for this pr!
@git-anurag-hub I didn't change docs in this PR, can you elaborate?

**Comment by JaeAeich** [comment]
Hey @git-anurag-hub, as @SalihuDickson has changed the build command to a script, this PR will not  add much to anything, so I am closing it.

**Pull request closed without merging**